### ISSUE TITLE
数据大小是均衡的，训练几万步，为什么显存越训越大，完全不合理

### ISSUE THREAD

**Issue by Wu-Xiang123** [modelscope/ms-swift#5747]
**Describe the bug**
What the bug is, and how to reproduce, better with screenshots(描述bug以及复现过程，最好有截图)
每个样本的大小基本是完全固定的，且通过max_length进行了峰值限制，为什么显存还是越训越大，直到oom？使用的是qwen2.5-vl-3b的多模态训练

**Your hardware and system info**
Write your system info like CUDA version/system/GPU/torch version here(在这里给出硬件信息和系统信息，如CUDA版本，系统，GPU型号和torch版本等)
amd mi308x机器，pytorch260。 ms-swfit-v3.8.0.dev0

**Additional context**
Add any other context about the problem here(在这里补充其他信息)

**Comment by Wu-Xiang123** [comment]
[显存异常logging.txt](https://github.com/user-attachments/files/22246199/logging.txt)
训练日志

**Comment by Wu-Xiang123** [comment]
启动命令：
args='--model models/Qwen/Qwen2___5-VL-3B-Instruct \
    --dataset xxx \
    --load_from_cache_file true \
    --new_special_tokens xxx \
    --lazy_tokenize true \
    --split_dataset_ratio 0 \
    --dataset_num_proc 4 \
    --dataloader_num_workers 4 \
    --do_train true \
    --do_eval false \
    --metric acc \
    --train_type full \
    --torch_dtype bfloat16 \
    --tuner_backend peft \
    --learning_rate 5e-5 \
    --lr_scheduler_type cosine \
    --warmup_ratio 0.02 \
    --freeze_vit false \
    --freeze_llm false \
    --freeze_aligner false \
    --num_train_epochs 1 \
    --per_device_train_batch_size 2 \
    --gradient_accumulation_steps 1 \
    --save_steps 1000 \
    --save_total_limit 100 \
    --logging_steps 10 \
    --max_length 6000 \
    --max_new_tokens 1000 \
    --output_dir xxx \
    --attn_impl flash_attn \
    --use_liger_kernel true '

**Comment by QixingJiang** [comment]
同问，理论上来说train_dataset的samples是随机的，但是我每次训练看log，每个logger_batch打印的显存都是逐渐上升的，不太理解这个原理

**Comment by hjh0119** [comment]
export PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True
export MAX_PIXELS=602112

**Comment by QixingJiang** [comment]
> 同问，理论上来说train_dataset的samples是随机的，但是我每次训练看log，每个logger_batch打印的显存都是逐渐上升的，不太理解这个原理

补充，MAX_PIXELS=602112是设置过的。

**Comment by Wu-Xiang123** [comment]
> expandable_segments

感谢回复！
MAX_PIXELS我已经设置过了500000。

我试下PYTORCH_CUDA_ALLOC_CONF

**Comment by Wu-Xiang123** [comment]
> > expandable_segments
> 
> 感谢回复！ MAX_PIXELS我已经设置过了500000。
> 
> 我试下PYTORCH_CUDA_ALLOC_CONF

两个都设置了还是同样的问题

<img width="1366" height="1992" alt="Image" src="https://github.com/user-attachments/assets/9516506c-6b5b-455a-9abb-159b361ee746" />

**Comment by hjh0119** [comment]
Sorry — I missed that your GPU is an AMD MI308. PYTORCH_CUDA_ALLOC_CONF only applies to the CUDA (NVIDIA) backend. Could you please try exporting the ROCm/HIP equivalent and retry:

`export PYTORCH_HIP_ALLOC_CONF=expandable_segments:True`

**Comment by Wu-Xiang123** [comment]
> Sorry — I missed that your GPU is an AMD MI308. PYTORCH_CUDA_ALLOC_CONF only applies to the CUDA (NVIDIA) backend. Could you please try exporting the ROCm/HIP equivalent and retry:
> 
> `export PYTORCH_HIP_ALLOC_CONF=expandable_segments:True`

[logging.txt](https://github.com/user-attachments/files/22251681/logging.txt)
改用了PYTORCH_HIP_ALLOC_CONF，还是一样的现象，奇怪。。。

**Comment by slin000111** [comment]
> > Sorry — I missed that your GPU is an AMD MI308. PYTORCH_CUDA_ALLOC_CONF only applies to the CUDA (NVIDIA) backend. Could you please try exporting the ROCm/HIP equivalent and retry:
> > `export PYTORCH_HIP_ALLOC_CONF=expandable_segments:True`
> 
> [logging.txt](https://github.com/user-attachments/files/22251681/logging.txt) 改用了PYTORCH_HIP_ALLOC_CONF，还是一样的现象，奇怪。。。

Remove `flash-attn` and `liger_kernel`, run the training test, and observe whether the video memory usage is normal. It's difficult to reproduce this issue without AMD GPUs.
